addon/addon-test-support/@ember/test-helpers/-internal/build-registry.ts

```diff
@@ -116,13 +120,15 @@ export default function (resolver: Resolver) {
 
   exposeRegistryMethodsWithoutDeprecations(container);
 
-  if (has('ember-data/setup-container')) {
+  if (macroCondition(dependencySatisfies('ember-data', '>= 2.3'))) {
```

Comment:
This only works if you declare a dependency (probably in this case an optional peer dependency) on ember-data.

Otherwise, a strict package manager like pnpm isn't going to be able to resolve ember-data from here, and the importSync would fail.